metals/src/main/scala/scala/meta/internal/process/SystemProcess.scala

```diff
@@ -1,30 +1,109 @@
 package scala.meta.internal.process
 
-import java.nio.file.Path
+import java.io.IOException
+import java.io.InputStream
+import java.util.concurrent.TimeUnit
 
-import scala.sys.process._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters._
+import scala.sys.process.BasicIO
+import scala.util.control.NonFatal
 
-import scala.meta.internal.metals.Time
-import scala.meta.internal.metals.Timer
-import scala.meta.pc.CancelToken
+import scala.meta.internal.ansi.AnsiFilter
+import scala.meta.io.AbsolutePath
+
+trait SystemProcess {
+  def complete: Future[Int]
+  def cancel: Unit
+}
 
 object SystemProcess {
+
   def run(
-      shortName: String,
-      args: List[String],
-      reproduceArgs: List[String],
-      cwd: Path,
-      token: CancelToken
-  ): Unit = {
-    val exportTimer = new Timer(Time.system)
-    scribe.info(args.mkString("process: ", " ", ""))
-    val exit = Process(args, cwd = Some(cwd.toFile())).!
-    if (exit != 0) {
-      val message = s"$shortName command failed with exit code $exit, " +
-        s"to reproduce run the command below:\n\t${reproduceArgs.mkString(" ")}"
-      throw new MessageOnlyException(message)
-    } else {
-      scribe.info(s"time: ran '$shortName' in $exportTimer")
+      cmd: List[String],
+      cwd: AbsolutePath,
+      redirectErrorOutput: Boolean,
+      env: Map[String, String]
+  ): SystemProcess = {
+
+    try {
+      val builder = new ProcessBuilder(cmd.asJava)
+      builder.directory(cwd.toNIO.toFile)
+      val envMap = builder.environment()
+      envMap.putAll(env.asJava)
```

Comment:
We should also add Java Home

Reply:
thx! Good catch. Added it in `ShellRunner`